src/esp/physics/BulletObject.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <Magnum/Trade/MeshData3D.h>
+#include <Magnum/Trade/MeshObjectData3D.h>
+#include <Corrade/Containers/Optional.h>
+#include <Corrade/Containers/Reference.h>
+#include <btBulletDynamicsCommon.h>
+#include "esp/assets/Asset.h"
+#include "esp/assets/BaseMesh.h"
+#include "esp/assets/MeshData.h"
+#include "esp/assets/GenericInstanceMeshData.h"
+#include "esp/assets/FRLInstanceMeshData.h"
+#include "esp/core/esp.h"
+
+namespace esp {
+namespace physics {
+
+class BulletRigidObject : public scene::SceneNode {
```

Comment:
We never ever inherit any class from the sceneNode to avoid "double deallocation", and to avoid ownership problems in pyBindings (if this class needs to be exposed in py which is very likely). This is because the SG memory management is automatically handled by magnum lib. Not so sure if the above theory still holds. If it is, then this is a problem here.

A "camera, agent, sensor, or light etc." is treated as an attached object to the scene node.
I believe the "rigid object" should follow the same logic.

More details (Pay attention to the figures in the PR description):
https://github.com/fairinternal/habitat-sim/pull/40

Reply:
This will get fully resolved with #64.